.travis.yml

```diff
@@ -6,52 +6,48 @@ dist: focal
 
 compiler:
   - gcc
+  - clang
+
+arch:
+  - amd64
+  - arm64
 
 env:
-  matrix:
-    - COMPILER=g++ CCOMPILER=gcc PUSH_COVERAGE=ON TRAVIS_BUILD_DOCS=ON
-    - COMPILER=g++-8 CCOMPILER=gcc-8
-    - COMPILER=g++-10 CCOMPILER=gcc-10
-    - COMPILER=clang++ CCOMPILER=clang
   global:
     - secure: "CQRYHWlg/WDu5DBUeDwGo+rPeOofN08DhiLUNlLtZjMWaRyP0Cop1qVaFs8ESOkYiWek2MdpvjZud+7hL+yx2ogvNx4SfHpUMCDKYgcX+YQ9MmYwabvoKq8N6KVXE3lbPp549TonHdDuNCWNKRniNjYtrij5J+IiIiT8/6Txo2p9RWk6YSUTdXJ9YrfuWMtRuF5uo9SHGyujv8pOJKedrwWoSBbHT44jnwfHMVe/C8jgjwlrJ9N3iXOtsG6sj+UTS8vOpL+jpBONEbBfHgSFU57I7IFNdPQbSObpVwG9geOAHT7IQQyQ9hp2AJoFxxVURB5SzqztDDpQ0XIF76vuH9tA/fF2pwDsLRmcLR8JU1TCmQgvnlYD0+Or9S1Dq0tQME5AP+21Hk2zVcGdbgQP7XWix758F0vpOXa4PXw8TmAjP2jKyAMHlzR3icr3+OmKSK3uXMMt2HSMOJQ+JvFxr//DM493i/VGyeY25/zu3A9RstiE+1d82Fi9xKOmMf4smvSkjOgT0b727jqNbNe6CvEKQUmqHabzYRQzUVz6WPVDHBxZP7AiKmZIVQXYnDsVXywStkSoxxY5En6XKpq0GR3bIVtUMORgZPoZi7Jni+/4EckcYH8g9mpsQf9tPRcOZ2WIvt5gqp2MZuwBLBRcbxihuECfBscqdeA0oDU5AZw="
     - GH_REPO_NAME: crow
     - DOXYFILE: $TRAVIS_BUILD_DIR/Doxyfile
     - GH_REPO_REF: github.com/mrozigor/crow.git
     - THEME_REPO_REF: github.com/mrozigor/darxygen.git
 
+
 addons:
   apt:
-    sources:
-      - ubuntu-toolchain-r-test
-      - boost-latest
-      - llvm-toolchain-precise
     packages:
-      - g++
-      - g++-8
-      - g++-10
-      - clang
       - libboost-all-dev
       - doxygen
       - doxygen-doc
       - doxygen-latex
       - doxygen-gui
       - graphviz
 
+before_install:
+  - if [ "$TRAVIS_COMPILER" == "gcc" -a "$TRAVIS_CPU_ARCH" == "amd64" ]; then export PUSH_COVERAGE=ON; fi
+  - if [ "$TRAVIS_BRANCH" == "master" -a "$TRAVIS_PULL_REQUEST" == "false" ]; then export TRAVIS_BUILD_DOCS=ON; fi
```

Comment:
We should also add here `"$TRAVIS_COMPILER" == "gcc" -a "$TRAVIS_CPU_ARCH" == "amd64"` so that Travis doesn't run generating docs 4 times (or maybe is there other, simpler way?).

Reply:
hmmm, we can add `"$PUSH_COVERAGE" == "ON"` since that implies `"$TRAVIS_COMPILER" == "gcc" -a "$TRAVIS_CPU_ARCH" == "amd64"`, I'll get on that